packages/flutter/lib/src/painting/borders.dart

```diff
@@ -533,6 +533,67 @@ abstract class ShapeBorder {
   ///  * [Path.contains], which can tell if an [Offset] is within a [Path].
   Path getInnerPath(Rect rect, { TextDirection? textDirection });
 
+  /// Paint in a given canvas with given paint.
+  ///
+  /// This works as an optimization for [getOuterPath] + [Canvas.drawPath],
+  /// which can be expensive.
+  /// When a shape provides its own [paintBox] that calls methods such as
+  /// [Canvas.drawRect] or [Canvas.drawCircle], drawing is faster.
+  ///
+  /// A standard paintBox implementation is canvas.draw followed by
+  /// what is inside [getOuterPath].
+  ///
+  /// As an example, see how [Path.addRect] became [Canvas.drawRect]:
+  ///
+  /// @override
```

Comment:
putting this in \`\`\`dart will give syntax highlighting in the API docs and (once my PR lands) will actually do code analysis automatically

Reply:
Done, thanks.